engine/classes/Elgg/Config.php

```diff
@@ -119,12 +141,24 @@ function get($name, $site_guid = 0) {
 			return null;
 		}
 	
-		if ($site_guid == $this->CONFIG->site_guid || $site_guid === null) {
-			$this->CONFIG->$name = $value;
+		if ($site_guid == $this->config->site_guid || $site_guid === null) {
+			$this->config->$name = $value;
 		}
 	
 		return $value;
 	}
+
+	/**
+	 * Get a config value for the current site if it's already loaded. This should be used instead of
+	 * reading directly from global $CONFIG.
+	 *
+	 * @param string $name Name of the configuration value
+	 *
+	 * @return mixed Returns null if value isn't set
+	 */
+	public function getVolatile($name) {
```

Comment:
I'm open to suggestions for a better name for this... I also considered adding another argument to get() to do this, but that seemed more messy.